tests/tests/test_compat.py

```diff
@@ -0,0 +1,291 @@
+    num_updated = 0
+    while num_updated < num_devices:
+        if datetime.now(timezone.utc) > deadline:
+            pytest.fail("timeout waiting for devices to submit inventory")
+
+        rsp = api_inventory.call(
+            "GET", inventory.URL_DEVICES, qs_params={"per_page": num_devices * 2},
+        )
+        assert rsp.status_code == 200
+        dev_invs = rsp.json()
+        assert (
+            len(dev_invs) <= num_devices
+        ), "Received more devices from inventory than there exists"
+        if len(dev_invs) < num_devices:
+            time.sleep(1)
+            continue
+        # Check if inventories has been updated since starting this loop
+        num_updated = 0
+        for device in dev_invs:
+            # datetime does not have an RFC3339 parser, but we can convert
+            # the timestamp to a compatible ISO format by removing
+            # fractions and replacing the Zulu timezone with GMT.
+            updated_ts = datetime.fromisoformat(
+                device["updated_ts"].split(".")[0] + "+00:00"
+            )
+            if updated_ts > update_after:
+                num_updated += 1
+            else:
+                time.sleep(1)
+                break
+
+
+def assert_successful_deployment(api_deployments, deployment_id, timeout=TIMEOUT):
+    """
+    Waits for the ongoing deployment (specified by deployment_id) to finish
+    and asserting all devices were successfully upgraded.
+    :param api_deployments: testutils.api.client.ApiClient setup and authorized
+                             to use the deployments management api,
+                             i.e. api_client.with_auth(api_token)
+    :param deployment_id: deployment id to watch
+    :param timeout: optional timeout value to wait for deployment (defaults to 5min)
+    """
+    deadline = datetime.now() + timeout
+    while True:
+        rsp = api_deployments.call(
+            "GET", deployments.URL_DEPLOYMENTS_ID.format(id=deployment_id)
+        )
+        assert rsp.status_code == 200
+
+        dpl = rsp.json()
+        if dpl["status"] == "finished":
+            rsp = api_deployments.call(
+                "GET", deployments.URL_DEPLOYMENTS_STATISTICS.format(id=deployment_id),
+            )
+            assert rsp.status_code == 200
+            assert rsp.json()["failure"] == 0
+            assert rsp.json()["success"] == dpl["device_count"]
+            break
+        elif datetime.now() > deadline:
+            pytest.fail("timeout: Waiting for devices to update")
+        else:
+            time.sleep(1)
+
+
+class TestMenderCompatibility:
+    """
+    This class contains compatibility tests for assessing server compatibility
+    with older clients.
+    """
+
+    def test_open_source_compatibility(self, setup_os_compat):
+        self.compatibility_test_impl(setup_os_compat)
+
+    def test_enterprise_compatibility(self, setup_ent_compat):
+        self.compatibility_test_impl(setup_ent_compat)
+
```

Comment:
There is a test suite selection in the integration tests. The enterprise ones need `Enterprise` in the class name.